backend/server/game_api.js

```diff
@@ -0,0 +1,637 @@
+const express = require('express');
+const gameRouter = express.Router();
+const {pool} = require('./db');
+
+
+// Middlewares
+gameRouter.use('/', (req, res, next) => {
+    if (!req.user) {
+        return res.status(401).json({msg: 'You need to login first'});
+    }
+    next();
+});
+
+
+gameRouter.use('/games/:game_id', async (req, res, next) => {
+    const game = await pool.query('select * from games where id = $1', [req.params.game_id]);
+    if (game.rows.length === 0) {
+        return res.status(400).json({msg: 'Game does not exist'});
+    }
+    next();
+});
+
+gameRouter.use('/games/:game_id', async (req, res, next) => {
+    const game = await pool.query('select * from games where id = $1', [req.params.game_id]);
+    if (game.rows[0].user1 !== req.user.id && game.rows[0].user2 !== req.user.id) {
+        return res.status(400).json({msg: 'User not part of game'});
+    }
+    next();
+});
+
+// Vars
+const shipsAmount = {
```

Comment:
This is not part of the Game API. Please move it to a dedicated file called `ships.js`.